kotlin/perform-spatial-operations/src/main/java/com/esri/arcgisruntime/sample/performspatialoperations/MainActivity.kt

```diff
@@ -0,0 +1,210 @@
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.GeometryEngine
+import com.esri.arcgisruntime.geometry.Part
+import com.esri.arcgisruntime.geometry.PartCollection
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.PointCollection
+import com.esri.arcgisruntime.geometry.Polygon
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val inputGeometryGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGeometryGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  // simple black line symbol for outlines
+  private val lineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 1f)
+  private val resultFillSymbol = SimpleFillSymbol(
+    SimpleFillSymbol.Style.SOLID, Color.RED, lineSymbol
+  )
+  private lateinit var inputPolygon1: Polygon
+  private lateinit var inputPolygon2: Polygon
+
+  // the spatial operation switching menu items.
+  private var noOperationMenuItem: MenuItem? = null
+  private var intersectionMenuItem: MenuItem? = null
+  private var unionMenuItem: MenuItem? = null
+  private var differenceMenuItem: MenuItem? = null
+  private var symmetricDifferenceMenuItem: MenuItem? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // create an ArcGISMap with a light gray canvas basemap
+      map = ArcGISMap(Basemap.createLightGrayCanvas())
+
+      // create graphics overlays to show the inputs and results of the spatial operation
+      graphicsOverlays.add(inputGeometryGraphicsOverlay)
+      graphicsOverlays.add(resultGeometryGraphicsOverlay)
+
+    }
+
+    // create input polygons and add graphics to display these polygons in an overlay
+    createPolygons()
+
+    // center the map view on the input geometries
+    val envelope = GeometryEngine.union(inputPolygon1, inputPolygon2).extent
+    mapView.setViewpointGeometryAsync(envelope, 20.0)
+  }
+
+  private fun showGeometry(resultGeometry: Geometry) {
+    // add a graphic from the result geometry, showing result in red (0xFFE91F1F)
+    val graphic = Graphic(resultGeometry, resultFillSymbol).also {
+      // select the result to highlight it
+      it.isSelected = true
+    }
+
+    resultGeometryGraphicsOverlay.graphics.add(graphic)
+  }
+
+  private fun createPolygons() {
+    // create input polygon 1
+    PointCollection(SpatialReferences.getWebMercator()).run {
```

Comment:
maybe change this to an apply like inner and outer ring? `inputPolygon1 = Polygon(PointCollection(SpatialRef)).apply {`

Reply:
sorry about missing that, should have seen it.